### ISSUE TITLE
Verify content(OpenPGP/MIME signed message) with detached signature

### ISSUE THREAD

**Issue by DenBond7** [vanitasvitae/pgpainless#240]
Hi @vanitasvitae

I've found an issue when I was trying to verify some content with a detached signature

![image](https://user-images.githubusercontent.com/2863246/153411210-e210d479-bd5f-4a47-95b4-dc7eff1b19f4.png)

**Original source**

```
To: default@flowcrypt.test
From: Den <denbond7@flowcrypt.test>
Subject: NOT encrypted + signed(detached)
Autocrypt: addr=denbond7@flowcrypt.test; keydata=
 mDMEYIucWBYJKwYBBAHaRw8BAQdAew+8mzMWyf3+Pfy49qa60uKV6e5os7de4TdZceAWUq+0
 F2RlbmJvbmQ3QGZsb3djcnlwdC50ZXN0iHgEExYKACAFAmCLnFgCGwMFFgIDAQAECwkIBwUV
 CgkICwIeAQIZAQAKCRDDIInNavjWzm3JAQCgFgCEyD58iEa/Rw/DYNoQNoZC1lhw1bxBiOcI
 btkdBgEAsDFZu3TBavOMKI7KW+vfMBHtRVbkMNpvunaAldoabgO4OARgi5xYEgorBgEEAZdV
 AQUBAQdAB1/Mrq5JGYim4KqGTSK4OESQUwPgK56q0yrkiU9WgyYDAQgHiHUEGBYKAB0FAmCL
 nFgCGwwFFgIDAQAECwkIBwUVCgkICwIeAQAKCRDDIInNavjWzjMgAQCU+R1fItqdY6lt9jXU
 qipmXuqVaEFPwNA8YJ1rIwDwVQEAyUc8162KWzA2iQB5akwLwNr/pLDDtOWwhLUkrBb3mAc=
Message-ID: <3fe297ba-3e53-7073-ee00-d0e516831264@flowcrypt.test>
Date: Tue, 8 Feb 2022 17:44:12 +0200
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101
 Thunderbird/68.7.0
MIME-Version: 1.0
Content-Type: multipart/signed; micalg=pgp-sha256;
 protocol="application/pgp-signature";
 boundary="4S1LR18a3JU2YiTxqubmlY1fr2BkmX171"

This is an OpenPGP/MIME signed message (RFC 4880 and 3156)
--4S1LR18a3JU2YiTxqubmlY1fr2BkmX171
Content-Type: multipart/mixed; boundary="OSR6TONWKJD9dgyc2XH5AQPNnAs7pdg1t"

--OSR6TONWKJD9dgyc2XH5AQPNnAs7pdg1t
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: quoted-printable
Content-Language: en-US

NOT encrypted + signed(detached)



--OSR6TONWKJD9dgyc2XH5AQPNnAs7pdg1t--

--4S1LR18a3JU2YiTxqubmlY1fr2BkmX171
Content-Type: application/pgp-signature; name="signature.asc"
Content-Description: OpenPGP digital signature
Content-Disposition: attachment; filename="signature.asc"

-----BEGIN PGP SIGNATURE-----

iHUEARYIAB0WIQTBZCjWAcs5N4nPYdTDIInNavjWzgUCYgKPzAAKCRDDIInNavjW
zmdoAP0TdFt1OWqosHhXxt2hNYqZQMc6bgQRpJNL029nRyzkPAD/SoYJ4T+aYEhw
11qrbXloqkr0G3QaA6/zk31RPMI/bgI=
=o5Ze
-----END PGP SIGNATURE-----

--4S1LR18a3JU2YiTxqubmlY1fr2BkmX171--

```

[mail.eml.txt](https://github.com/pgpainless/pgpainless/files/8040998/mail.eml.txt)

**Code**

```java
package org.pgpainless.decryption_verification;

import static org.junit.jupiter.api.Assertions.assertTrue;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.nio.charset.StandardCharsets;

import org.bouncycastle.openpgp.PGPException;
import org.bouncycastle.util.io.Streams;
import org.junit.jupiter.api.Test;
import org.pgpainless.PGPainless;
import org.pgpainless.decryption_verification.cleartext_signatures.InMemoryMultiPassStrategy;

public class VerifyDetachedSignatureTest {

    @Test
    public void verify() throws PGPException, IOException {
        String signedContent = "Content-Type: multipart/mixed; boundary=\"OSR6TONWKJD9dgyc2XH5AQPNnAs7pdg1t\"\n" +
                "\n" +
                "--OSR6TONWKJD9dgyc2XH5AQPNnAs7pdg1t\n" +
                "Content-Type: text/plain; charset=utf-8\n" +
                "Content-Transfer-Encoding: quoted-printable\n" +
                "Content-Language: en-US\n" +
                "\n" +
                "NOT encrypted + signed(detached)\n" +
                "\n" +
                "\n" +
                "\n" +
                "--OSR6TONWKJD9dgyc2XH5AQPNnAs7pdg1t--\n";
        String signature = "-----BEGIN PGP SIGNATURE-----\n" +
                "\n" +
                "iHUEARYIAB0WIQTBZCjWAcs5N4nPYdTDIInNavjWzgUCYgKPzAAKCRDDIInNavjW\n" +
                "zmdoAP0TdFt1OWqosHhXxt2hNYqZQMc6bgQRpJNL029nRyzkPAD/SoYJ4T+aYEhw\n" +
                "11qrbXloqkr0G3QaA6/zk31RPMI/bgI=\n" +
                "=o5Ze\n" +
                "-----END PGP SIGNATURE-----\n";
        String pubkey = "-----BEGIN PGP PUBLIC KEY BLOCK-----\n" +
                "Version: PGPainless\n" +
                "\n" +
                "mDMEYIucWBYJKwYBBAHaRw8BAQdAew+8mzMWyf3+Pfy49qa60uKV6e5os7de4TdZ\n" +
                "ceAWUq+0F2RlbmJvbmQ3QGZsb3djcnlwdC50ZXN0iHgEExYKACAFAmCLnFgCGwMF\n" +
                "FgIDAQAECwkIBwUVCgkICwIeAQIZAQAKCRDDIInNavjWzm3JAQCgFgCEyD58iEa/\n" +
                "Rw/DYNoQNoZC1lhw1bxBiOcIbtkdBgEAsDFZu3TBavOMKI7KW+vfMBHtRVbkMNpv\n" +
                "unaAldoabgO4OARgi5xYEgorBgEEAZdVAQUBAQdAB1/Mrq5JGYim4KqGTSK4OESQ\n" +
                "UwPgK56q0yrkiU9WgyYDAQgHiHUEGBYKAB0FAmCLnFgCGwwFFgIDAQAECwkIBwUV\n" +
                "CgkICwIeAQAKCRDDIInNavjWzjMgAQCU+R1fItqdY6lt9jXUqipmXuqVaEFPwNA8\n" +
                "YJ1rIwDwVQEAyUc8162KWzA2iQB5akwLwNr/pLDDtOWwhLUkrBb3mAc=\n" +
                "=pXF6\n" +
                "-----END PGP PUBLIC KEY BLOCK-----\n";


        DecryptionStream verifier = PGPainless.decryptAndOrVerify()
                .onInputStream(new ByteArrayInputStream(signedContent.getBytes(StandardCharsets.UTF_8)))
                .withOptions(
                        new ConsumerOptions()
                                .addVerificationOfDetachedSignatures(new ByteArrayInputStream(signature.getBytes(StandardCharsets.UTF_8)))
                                .addVerificationCerts(PGPainless.readKeyRing().keyRingCollection(pubkey, true).getPgpPublicKeyRingCollection())
                                .setMultiPassStrategy(new InMemoryMultiPassStrategy())
                );

        Streams.drain(verifier);
        verifier.close();
        OpenPgpMetadata metadata = verifier.getResult();
        assertTrue(metadata.isVerified());
    }
}
```
[VerifyDetachedSignatureTest.java.txt](https://github.com/pgpainless/pgpainless/files/8041014/VerifyDetachedSignatureTest.java.txt)

It seems `PGPainless` doesn't like `signedContent`

Just in case I'd like to add I can verify the content with `gpg` via `gpg --verify doc.sig doc`

**Comment by vanitasvitae** [comment]
I will take a look 👍

**Comment by vanitasvitae** [comment]
This is fixed with release 1.0.3+
